src/node.cc

```diff
@@ -2708,90 +2725,160 @@ struct DLib {
     uv_dlclose(&lib_);
   }
 #endif  // !__POSIX__
+
+  DLib(const DLib& other) = delete;
+  DLib(DLib&& other) = delete;
+  DLib& operator=(const DLib& other) = delete;
+  DLib& operator=(DLib&& other) = delete;
+
+  void AddEnvironment(Environment* env) {
+    if (users_.count(env) > 0) return;
```

Comment:
Would access to `users_` need to be protected by a mutex?

Reply:
It is kinda protected by `dlib_mutex`, because this method is only called from `DLOpen()`, but it would be nice to actually restrict access to the method so that it couldn't be called from another place.